server/src/com/thoughtworks/go/server/service/PluginService.java

```diff
@@ -160,6 +161,21 @@ private void update(Username currentUser, LocalizedOperationResult result, Plugi
         }
     }
 
+    private void notifyPluginSettingsChange(PluginSettings pluginSettings) {
+        String pluginId = pluginSettings.getPluginId();
+
+        for (GoPluginExtension extension : extensions) {
+            if (extension.canHandlePlugin(pluginId)) {
+                try {
+                    extension.notifyPluginSettingsChange(pluginId, pluginSettings.getSettingsMap());
+                    break;
```

Comment:
I think `break` should be outside of try block or It should be in finally to break loop once `extension.canHandlePlugin(pluginId)` you have extension for a pluginId.